osu.Game/Screens/Edit/Compose/Components/BlueprintContainer.cs

```diff
@@ -480,52 +431,47 @@ private bool moveCurrentSelection(DragEvent e)
             if (movementBlueprints == null)
                 return false;
 
-            if (snapProvider == null)
-                return true;
-
             Debug.Assert(movementBlueprintOriginalPositions != null);
 
             Vector2 distanceTravelled = e.ScreenSpaceMousePosition - e.ScreenSpaceMouseDownPosition;
 
-            // check for positional snap for every object in selection (for things like object-object snapping)
-            for (var i = 0; i < movementBlueprintOriginalPositions.Length; i++)
+            if (snapProvider != null)
             {
-                var testPosition = movementBlueprintOriginalPositions[i] + distanceTravelled;
+                // check for positional snap for every object in selection (for things like object-object snapping)
+                for (var i = 0; i < movementBlueprintOriginalPositions.Length; i++)
+                {
+                    var testPosition = movementBlueprintOriginalPositions[i] + distanceTravelled;
 
-                var positionalResult = snapProvider.SnapScreenSpacePositionToValidPosition(testPosition);
+                    var positionalResult = snapProvider.SnapScreenSpacePositionToValidPosition(testPosition);
 
-                if (positionalResult.ScreenSpacePosition == testPosition) continue;
+                    if (positionalResult.ScreenSpacePosition == testPosition) continue;
 
-                // attempt to move the objects, and abort any time based snapping if we can.
-                if (SelectionHandler.HandleMovement(new MoveSelectionEvent(movementBlueprints[i], positionalResult.ScreenSpacePosition)))
-                    return true;
+                    // attempt to move the objects, and abort any time based snapping if we can.
+                    if (SelectionHandler.HandleMovement(new MoveSelectionEvent<T>(movementBlueprints[i], positionalResult.ScreenSpacePosition)))
+                        return true;
+                }
             }
```

Comment:
yes, it's intended. all editors will have grids and snapping.

Reply:
I raised this because `IPositionSnapProvider` is currently coupled with time, so I wasn't sure at all if time is a valid thing in this level, but I guess it's fine if the intention is to reuse that interface here as well.

Reply:
The time part can just be ignored. We could refactor if time is found but it works fine as-is.